src/addons/mod/data/components/action/action.ts

```diff
@@ -139,4 +142,51 @@ export class AddonModDataActionComponent implements OnInit {
         CoreEvents.trigger(AddonModDataProvider.ENTRY_CHANGED, { dataId: dataId, entryId: entryId }, this.siteId);
     }
 
+    async actionsMenu(): Promise<void> {
+        const items = [
+            {
+                action: async () => {
+                    this.editEntry();
+                    await PopoverController.dismiss();
+                },
+                text: 'core.edit',
+                icon: 'fas-pen',
+            },
+            {
+                action: async () => {
+                    this.deleteEntry();
+                    await PopoverController.dismiss();
+                },
+                text: 'core.delete',
+                icon: 'fas-trash',
+            },
+            {
+                action: async () => {
+                    this.disapproveEntry();
+                    await PopoverController.dismiss();
+                },
+                text: 'addon.mod_data.disapprove',
+                icon: 'fas-rotate-left',
+            },
+        ];
+
+        if (this.mode === AddonModDataTemplateMode.LIST) {
+            items.unshift({
+                action: async () => {
+                    this.viewEntry();
+                    await PopoverController.dismiss();
+                },
+                text: 'addon.mod_data.more',
+                icon: 'fas-magnifying-glass-plus',
+            });
+        }
+
+        await CoreDomUtils.openPopover<AddonModDataActionsMenuItem>({
```

Comment:
```suggestion
        await CoreDomUtils.openPopover({
```

The type passed to `openPopover` is the type of the value passed to `PopoverController.dismiss()`. But you aren't passing any value when dismissing the popover, so you don't need to define this type (it's void by default).